### PR TITLE
Remove inline js of com_cache

### PR DESCRIPTION
**Pull request by laoneo**
Remove inline js of com_cache. Not sure why the transpiler made _this in the head.

### CODE REVIEW

File: media/com_cache/js/admin-cache-default.es6.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+Joomla = window.Joomla || {};
+
+((document, Joomla) => {
+  'use strict';
+
+  document.addEventListener('DOMContentLoaded', () => {
+    [].slice.call(document.querySelectorAll('.cache-entry')).forEach((el) => {
+      el.addEventListener('click', () => {
+        Joomla.isChecked(this.checked);
```
Comment (dneukirchen) [inline media/com_cache/js/admin-cache-default.es6.js]: this in this case is the window object and not the element (arrow function).
You can use Joomla.isChecked(event.currentTarget.checked)

**Review by dneukirchen — CHANGES_REQUESTED** [review]

File: media/com_cache/js/admin-cache-default.es6.js (hunk 1 above)
Comment (laoneo) [reply media/com_cache/js/admin-cache-default.es6.js]: Ok. Why does it work then? Is this code needed at all?

Comment (dneukirchen) [reply media/com_cache/js/admin-cache-default.es6.js]: It does not work as it should. `<input type="hidden" name="boxchecked">` is wrongly decremented with every click. 

boxchecked count is needed for the toolbar button when we check if any item is selected before deletion. 

reproduce:

- create folder in cache directory
- open cache manager in backend
- check item
- uncheck it
- click delete 
- Dont see "Please first make a selection from the list."

Comment (C-Lodder) [reply media/com_cache/js/admin-cache-default.es6.js]: @dneukirchen If I'm not mistaken, `this` is the same as `event.currentTarget`

Comment (dneukirchen) [reply media/com_cache/js/admin-cache-default.es6.js]: Not in arrow functions. The value of `this` inside the arrow function is determined by where the arrow function is defined, not where it is used (the element)

Comment (laoneo) [reply media/com_cache/js/admin-cache-default.es6.js]: This explains the strange code on the top of the normal js file with _this. Fixed.

**Review by dneukirchen — APPROVED** [review]

**Pull request merged**